### DISCUSSION TITLE
How to get access to route params from validation?

### DISCUSSION THREAD

**Discussion by bradleybernard** [spatie/laravel-data#286] [Q&A]
Usually in my form requests I'll run a model policy by:

```
public function authorize(): bool
{
    return $this->user()->can(
          'delete',
          $this->route('myModel'),
      );
}
```

How can I do that with the static method authorize available? I dug into src and it looks like authorize is called with no params, so I wasn't able to access  the Request $request object, since it doesn't resolve from container.

Any advice here? Right now I'd have to have a form request + laravel-data model file, which is a lot.

I know laravel-data is supposed to be more "abstract" but having a form where you use laravel-data to edit the model, it makes sense to have a form request like way to validate authorization to that model. Curious if I missed something in the docs or someone has a solution for this! 

Any help would be appreciated. Thank you! :)

**Comment by Nir-An** [comment]
Hi @bradleybernard  :)

if I understand correctlly, the  ``$this->user`` reffer to the ``user`` that is already ``authorized`` 

soo you can do the same thing with the ``Auth`` helper
Docs: [request-to-data-object - authorizing-a-request](https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/request-to-data-object#content-authorizing-a-request)


```php

use Illuminate\Http\Request;

class SongData extends Data
{
    public function __construct(
        public string $title,
        public string $artist,
    ) {
    }

   public function authorize(): bool
   {
     return Auth::user()->can(
        'delete',
         Request::route('myModel')
       );
   }
}
```


If I am wrong please tell me

https://spatie.be/docs/laravel-data/v2/as-a-data-transfer-object/request-to-data-object

**Reply by bradleybernard** [reply]
Hey @Nir-An! `Auth::user()` can indeed get me the logged in `User` but how would I access route model binded params? 

IE: this code from a `FormRequest` class
```php
$this->route('myModel')
```

**Reply by rubenvanassche** [reply]
Use `request()->route('myModel')`

**Reply by scanningdeer** [reply]
If we could customize the normalizer used for routes (we can if we extend the pipeline function on each Data), we could do something like 
```
class RequestNormalizer implements Normalizer
{
    public function normalize(mixed $value): ?array
    {
        if (! $value instanceof Request) {
            return null;
        }

        return array_merge($value->route()->parameters(), $value->toArray());
    }
}
``` 

Example of adding via pipeline() 
```
 public static function pipeline(): DataPipeline
    {
        return parent::pipeline()->normalizer(
            new RequestNormalizer()
        );
    }
```

This is probably a simplistic view of how to use route parameters, but I think this is close to how I'd want/expect it to work.  If the normalizer was allowed to look at the DataClass it'd probably be possible to only pull requested parameters in.

**Reply by rubenvanassche** [reply]
I think adding a new pipeline before all other pipelines which injects the route parameters based upon PHP attributes is a more sane approach since you can do both in the same step.
